functions/Update-DbaInstance.ps1

```diff
@@ -289,13 +300,96 @@ function Update-DbaInstance {
                 Write-Message -Level Warning -Message "Update failed: $($_.Notes -join ' | ')"
             }
         }
+        function Join-AdminUnc {
+            <#
+                .SYNOPSIS
+                Internal function. Parses a path to make it an admin UNC.
+            #>
+            [CmdletBinding()]
+            param (
+                [Parameter(Mandatory)]
+                [ValidateNotNullOrEmpty()]
+                [DbaInstanceParameter]$ComputerName,
+
+                [Parameter(Mandatory)]
+                [ValidateNotNullOrEmpty()]
+                [string]$Path
+
+            )
+            if ($Path.StartsWith("\\")) {
+                return $filepath
+            }
+
+            $servername = $ComputerName.ComputerName
+            $newpath = Join-Path "\\$servername\" $Path.replace(':', '$')
+            return $newpath
+        }
+        function Copy-UncFile {
+            <#
+
+                SYNOPSIS
+                Internal function. Uses PSDrive to copy file to the remote system.
+
+                #>
+            param (
+                [Parameter(Mandatory)]
+                [ValidateNotNullOrEmpty()]
+                [DbaInstanceParameter]$ComputerName,
+
+                [Parameter(Mandatory)]
+                [ValidateNotNullOrEmpty()]
+                [string]$Path,
+
+                [Parameter(Mandatory)]
+                [ValidateNotNullOrEmpty()]
+                [string]$Destination,
+
+                [PSCredential]$Credential
+            )
+            if (([DbaInstanceParameter]$groupItem.ComputerName).IsLocalHost) {
+                $remoteFolder = $Destination
+            } else {
+                $uncFileName = Join-AdminUnc -ComputerName $ComputerName -Path $Destination
+                $driveSplat = @{
+                    Name       = 'UpdateCopy'
+                    Root       = $uncFileName
+                    PSProvider = 'FileSystem'
+                }
+                if ($Credential) { $driveSplat.Credential = $Credential }
```

Comment:
This is not required if you are going to splat the command. It handles dealing with `$Credential` not being provided.

Reply:
Sounds good, moved Credential to the splat. Didn't expect a native cmdlet to handle $null credentials properly.